internal/xpkg/fetch.go

```diff
@@ -143,6 +157,26 @@ func (i *K8sFetcher) Fetch(ctx context.Context, ref name.Reference, secrets ...s
 	if err != nil {
 		return nil, err
 	}
+	if len(i.validationPem) != 0 {
+		opts := []remote.Option{
+			remote.WithAuthFromKeychain(auth),
+			remote.WithContext(ctx),
+		}
+		pk, err := sigs.LoadPublicKeyRaw([]byte(i.validationPem), crypto.SHA256)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
@jessesanford can we pull this out to where we initially read in the PEM encoded pub key? Ideally we would fail fast if we get an error on this operation, and we don't need to load on every invocation of `Fetch()`